File: src/main/java/com/adinsight/agent/model/ToolCall.java
package com.adinsight.agent.model;

import java.util.LinkedHashMap;
import java.util.Map;

public record ToolCall(String id, String name, String argumentsJson) {
    public Map<String, Object> toOpenAiMap() {
        Map<String, Object> function = new LinkedHashMap<>();
        function.put("name", name);
        function.put("arguments", argumentsJson == null ? "{}" : argumentsJson);

        Map<String, Object> value = new LinkedHashMap<>();
        value.put("id", id);
        value.put("type", "function");
        value.put("function", function);
        return value;
    }
}

File: src/model/tool-call.ts
export type ToolCall = Readonly<{
  id: string;
  name: string;
  argumentsJson: string;
}>;

export function toolCall(input: {
  id: string;
  name: string;
  argumentsJson?: string | null;
}): ToolCall {
  return {
    id: input.id,
    name: input.name,
    argumentsJson: input.argumentsJson == null || input.argumentsJson.trim() === ""
      ? "{}"
      : input.argumentsJson
  };
}
